cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStatusTest.java

```diff
@@ -0,0 +1,200 @@
+        brokerOneBefore.put(CruiseControlLoadParameters.REPLICAS.getCruiseControlKey(), 10);
+        brokerLoadBeforeArray.add(brokerOneBefore);
+        brokersBeforeObject.put(CruiseControlRebalanceKeys.BROKERS.getKey(), brokerLoadBeforeArray);
+
+        JsonObject brokersAfterObject = new JsonObject();
+        JsonArray brokerLoadAfterArray = new JsonArray();
+        JsonObject brokerOneAfter = new JsonObject();
+        brokerOneAfter.put(CruiseControlRebalanceKeys.BROKER_ID.getKey(), BROKER_ONE_KEY);
+        brokerOneAfter.put(CruiseControlLoadParameters.CPU_PERCENTAGE.getCruiseControlKey(), 20.0);
+        brokerOneAfter.put(CruiseControlLoadParameters.REPLICAS.getCruiseControlKey(), 5);
+        brokerLoadAfterArray.add(brokerOneAfter);
+        brokersAfterObject.put(CruiseControlRebalanceKeys.BROKERS.getKey(), brokerLoadAfterArray);
+
+        proposal.put(CruiseControlRebalanceKeys.SUMMARY.getKey(), summary);
+        proposal.put(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey(), brokersBeforeObject);
+        proposal.put(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey(), brokersAfterObject);
+
+        return proposal;
+
+    }
+
+    @Test
+    public void testLoadParamExtract() {
+
+        JsonObject proposal = buildOptimizationProposal();
+
+        JsonArray loadBeforeArray = proposal.getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey())
+                .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+
+        Map<Integer, Map<String, Object>> output = KafkaRebalanceAssemblyOperator.extractLoadParameters(loadBeforeArray);
+
+        assertThat(output, hasKey(BROKER_ONE_KEY));
+        assertThat(output.get(BROKER_ONE_KEY), hasEntry(CruiseControlLoadParameters.CPU_PERCENTAGE.getKafkaRebalanceStatusKey(), 10.0));
+        assertThat(output.get(BROKER_ONE_KEY), hasEntry(CruiseControlLoadParameters.REPLICAS.getKafkaRebalanceStatusKey(), 10));
+
+    }
+
+    @Test
+    public void testCreateLoadMap() {
+
+        JsonObject proposal = buildOptimizationProposal();
+
+        JsonArray loadBeforeArray = proposal.getJsonObject(CruiseControlRebalanceKeys.LOAD_BEFORE_OPTIMIZATION.getKey())
+                .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+        JsonArray loadAfterArray = proposal.getJsonObject(CruiseControlRebalanceKeys.LOAD_AFTER_OPTIMIZATION.getKey())
+                .getJsonArray(CruiseControlRebalanceKeys.BROKERS.getKey());
+
+        JsonObject output = KafkaRebalanceAssemblyOperator.parseLoadStats(
+                loadBeforeArray, loadAfterArray);
+
+        assertThat(output.getMap(), hasKey("1"));
+
+        assertThat(output.getJsonObject("1").getMap(), hasKey(CruiseControlLoadParameters.REPLICAS.getKafkaRebalanceStatusKey()));
+
+        JsonObject replicas = output.getJsonObject("1").getJsonObject("replicas");
+
+        ArrayList<Integer> replicasList = new ArrayList<>();
```

Comment:
Why do we need an Arraylist here? Why not just do: `assertThat(replicas.getInteger("before"), is(10));` below?